mewarpx/mewarpx/diags_store/field_diagnostic.py

```diff
@@ -70,24 +84,65 @@ def plot_parameters(self):
 
             for param in self.plot_data_list:
                 if param == 'rho':
-                    data = np.array(mwxrun.get_gathered_rho_grid(include_ghosts=False)[0])
-
-                    fig, ax = plt.subplots(1, 1, figsize=(14, 14))
-                    plotting.ArrayPlot(
-                        array=data[:, :, 0],
-                        template='rho', xaxis='z', yaxis='x', ax=ax,
-                        draw_image=True, draw_contourlines=False,
-                        cmap='viridis', plot_name="rho_" + plot_name
-                    )
+                    data = mwxrun.get_gathered_rho_grid(include_ghosts=False)
+                    if mwxrun.me == 0:
+                        data = np.array(data[0])
+
+                        fig, ax = plt.subplots(1, 1, figsize=(14, 14))
+                        plotting.ArrayPlot(
+                            array=data[:, :, 0],
+                            template='rho', xaxis='z', yaxis='x', ax=ax,
+                            plot_name="rho_" + plot_name
+                        )
                 elif param == 'phi':
-                    data = np.array(mwxrun.get_gathered_phi_grid(include_ghosts=False)[0])
-
-                    fig, ax = plt.subplots(1, 1, figsize=(14, 14))
-                    plotting.ArrayPlot(
-                        array=data,
-                        template='phi', xaxis='z', yaxis='x', ax=ax,
-                        draw_image=True, draw_contourlines=False,
-                        cmap='viridis', plot_name="phi_" + plot_name
-                    )
+                    data = mwxrun.get_gathered_phi_grid(include_ghosts=False)
+                    if mwxrun.me == 0:
+                        data = np.array(data[0])
+
+                        fig, ax = plt.subplots(1, 1, figsize=(14, 14))
+                        plotting.ArrayPlot(
+                            array=data,
+                            template='phi', xaxis='z', yaxis='x', ax=ax,
+                            draw_image=True, draw_contourlines=False,
+                            cmap='viridis', plot_name="phi_" + plot_name
+                        )
                 else:
                     print("Attempted to plot a parameter for which plotting has not been implemented:", param)
+
+    def check_for_end_of_sim(self):
+        if mwxrun.get_it() == mwxrun.simulation.max_steps:
+            self.do_post_processing()
+
+    def do_post_processing(self):
+        if mwxrun.me == 0:
+            constants = picmi.constants
+            data_dirs = glob.glob('diags/diags*')
+
+            if len(data_dirs) == 0:
+                raise RuntimeError("No data files found.")
+
+            for datafolder in data_dirs:
+                if not "old" in datafolder:
+
+                    with pytest.warns(ResourceWarning):
```

Comment:
This wrapper should be in the necessary place in the test in `tests`. pytest shouldn't touch the core code at all.